File: source_funcs/go_transcoder/FIND_THE_ELEMENT_THAT_APPEARS_ONCE/FIND_THE_ELEMENT_THAT_APPEARS_ONCE.go
package main

func f_gold(arr []int, n int) int {
	var (
		ones            int = 0
		twos            int = 0
		common_bit_mask int
	)
	for i := int(0); i < n; i++ {
		twos = twos | (ones & arr[i])
		ones = ones ^ arr[i]
		common_bit_mask = ^(ones & twos)
		ones &= common_bit_mask
		twos &= common_bit_mask
	}
	return ones
}

File: rust_funcs/cpp_transcoder/FIND_THE_ELEMENT_THAT_APPEARS_ONCE/FIND_THE_ELEMENT_THAT_APPEARS_ONCE.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(arr: [i32;2], n: i32) -> i32 {  
    let mut ones = 0;  
    let mut twos = 0; 
    
    for i in 0..n {
        twos = twos | (ones & arr[i as usize]);
        ones = ones ^ arr[i as usize];
        let common_bit_mask = !(ones & twos);
        ones &= common_bit_mask;
        twos &= common_bit_mask;
    }
    ones  
}
